Sources/Polyline/CoreLocation.swift

```diff
@@ -0,0 +1,60 @@
+// Polyline.swift
+//
+// Copyright (c) 2015 Raphaël Mor
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+
+
+#if !canImport(CoreLocation)
+public typealias CLLocationDegrees = Double
+public typealias CLLocationSpeed = Double
+public typealias CLLocationDirection = Double
+public typealias CLLocationDistance = Double
+public typealias CLLocationAccuracy = Double
+
+public struct CLLocationCoordinate2D {
+    public let latitude: CLLocationDegrees
+    public let longitude: CLLocationDegrees
+}
+
+public struct CLLocation {
```

Comment:
Polyline might not be the only library in a Linux application that needs to work with coordinates. If the application also installs [Turf](https://github.com/mapbox/turf-swift/), both will have CLLocations and CLLocationCoordinate2Ds. It won’t result in compilation errors, but having two partial implementations of Core Location’s API can be confusing.

Instead, how about taking the “CL” prefix off these types and using them internally instead of Core Location types? For the public API, we can continue to provide Core Location–compatible methods (as long as Core Location can be imported), but behind the scenes they’d be pass-throughs to non–Core Location implementations. It’d be like the MapKit convenience methods.

Or perhaps we could define a protocol that both this class and the real CLLocation could conform to.

Reply:
Different concrete type internally would have an impact on the performance since it would have to be transformed yet another time. The protocol approach sounds viable.

Reply:
I took the simplest possible approach of making the public API use a uniquely named type that on Apple platforms is aliased to `CLLocationCoordinate2D` but on Linux platforms is a concrete struct definition. This is similar to the workarounds often found when deploying backwards to an older version of iOS that lacks certain Apple APIs. It should have no performance implications. The only drawbacks I can think of are that users have to dig a little deeper to realize they can use certain methods with `CLLocationCoordinate2D`, and that Linux clients will need to implement their own class to wrap `CLLocationCoordinate2D` in the unlikely event that they need one.